unstructured_ingest/v2/processes/connectors/zendesk/zendesk.py

```diff
@@ -0,0 +1,641 @@
+import bs4
+from pydantic import BaseModel, Field, Secret
+
+from unstructured_ingest.error import (
+    SourceConnectionError,
+)
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    BatchFileData,
+    BatchItem,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+from .wrapper import ZendeskArticle, ZendeskClient, ZendeskTicket
+
+CONNECTOR_TYPE = "zendesk"
+
+
+class ZendeskAdditionalMetadata(BaseModel):
+    item_type: str
+    leading_id: str  # is the same as id just being verbose.
+    tail_id: str  # last id in the batch.
+
+
+class ZendeskFileDataSourceMetadata(FileDataSourceMetadata):
+    """
+    inherits metadata object as tickets and articles
+    are treated in single batch, we need to denote indices ticket/article
+    as the source metadata.
+    """
+
+
+class ZendeskBatchFileData(BatchFileData):
+    additional_metadata: ZendeskAdditionalMetadata
+
+
+class ZendeskAccessConfig(AccessConfig):
+    api_token: str = Field(
+        description="API token for zendesk generated under Apps and Integrations"
+    )
+
+
+class ZendeskBatchItemTicket(BatchItem):
+    subject: str
+    description: str
+    item_type: str = "tickets"  # placeholder for downloader
+
+
+class ZendeskBatchItemArticle(BatchItem):
+    title: str
+    author_id: str
+    title: str
+    content: str
+
+
+class ZendeskConnectionConfig(ConnectionConfig):
+    subdomain: str = Field(description="Subdomain for zendesk site, <sub-domain>.company.com")
+    email: str = Field(description="Email for zendesk site registered at the subdomain")
+    access_config: Secret[ZendeskAccessConfig]
+
+    @contextmanager
+    def get_client(self) -> Generator["ZendeskClient", None, None]:
+
+        access_config = self.access_config.get_secret_value()
+
+        options = {
+            "email": self.email,
+            "subdomain": self.subdomain,
+            "token": access_config.api_token,
+        }
+
+        client = ZendeskClient(**options)
+        yield client
+
+    @asynccontextmanager
+    async def get_client_async(self) -> AsyncGenerator["ZendeskClient", None]:
+        """Provides an async context manager for ZendeskClient."""
+        access_config = self.access_config.get_secret_value()
+
+        options = {
+            "email": self.email,
+            "subdomain": self.subdomain,
+            "token": access_config.api_token,
+        }
+
+        client = ZendeskClient(**options)
+        yield client
+
+
+class ZendeskIndexerConfig(IndexerConfig):
+    batch_size: int = Field(
+        default=1,
+        description="Number of tickets or articles.",
+    )
+    item_type: str = Field(
```

Comment:
If we know it can only be a subset of strings, we can user `Literal` to add that for type checking. 
```suggestion
    item_type: Literal["tickets", "articles"] = Field(
```

Reply:
Also do we not want to support pulling in all this data in a single call? Otherwise they need to run 2 different pipelines if they want to ingest tickets and articles.